byterun/weak.c

```diff
@@ -71,6 +94,14 @@ CAMLprim value caml_ephe_create (value len)
   return res;
 }
 
+CAMLprim value caml_ephe_create (value len)
+{
+  intnat size = Long_val(len);
+  if (size < 0 || size > (Max_wosize - CAML_EPHE_FIRST_KEY))
+    caml_invalid_argument ("Weak.create");
```

Comment:
Wrong function name.
Also, it would be clearer if you defined a constant which was equal to Max_wosize - CAML_EPHE_FIRST_KEY.